App/automotive/src/main/java/com/amos/infotaimos/viewmodel/VehiclePropertiesPageViewModel.kt

```diff
@@ -28,7 +48,88 @@ class VehiclePropertiesPageViewModel : ViewModel() {
     fun sendNotification(context: Context, percentage: String, delay: Long) {
         Log.d("VM", "$percentage $delay")
         Timer().schedule(delay) {
-            NotificationManager.createNotification(context, context.getString(R.string.battery_notification,percentage))
+            NotificationManager.createNotification(
+                context,
+                context.getString(R.string.battery_notification, percentage)
+            )
+        }
+    }
+
+    private fun checkBatteryPermission(context: Context): Boolean {
+        return ContextCompat.checkSelfPermission(
+            context,
+            Car.PERMISSION_ENERGY
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    fun getBatteryPermission(context: Context, activity: Activity, view: View): Boolean {
+        batteryPermission = checkBatteryPermission(context)
+        if (!batteryPermission) {
+            ActivityCompat.requestPermissions(activity, arrayOf(Car.PERMISSION_ENERGY), 0)
+            batteryPermission = checkBatteryPermission(context)
```

Comment:
das funktioniert halt nicht weil das feedback von requestPermission nicht sofort kommt, sondern erst wenn der Nutzer reagiert

Reply:
Okay ich hab das requesten der Energy Permission jetzt in die main activity verschoben. Jetzt funktioniert das ganze auch richtig. Danke nochmal :D